pkg/controller/elasticsearch/nodespec/statefulset.go

```diff
@@ -118,6 +116,29 @@ func BuildStatefulSet(
 	return sset, nil
 }
 
+func setVolumeClaimsControllerReference(
+	persistentVolumeClaims []corev1.PersistentVolumeClaim,
+	es v1beta1.Elasticsearch,
+	scheme *runtime.Scheme,
+) ([]corev1.PersistentVolumeClaim, error) {
+	// set the owner reference of all volume claims to the ES resource,
+	// so PVC get deleted automatically upon Elasticsearch resource deletion
+	claims := make([]corev1.PersistentVolumeClaim, 0, len(persistentVolumeClaims))
+	for _, claim := range persistentVolumeClaims {
+		if err := controllerutil.SetControllerReference(&es, &claim, scheme); err != nil {
+			return nil, err
+		}
+		// Set block owner deletion to false as the statefulset controller might not be able to do that
+		// See https://github.com/elastic/cloud-on-k8s/issues/1884
+		refs := claim.OwnerReferences
+		for i := range refs {
+			refs[i].BlockOwnerDeletion = &f
```

Comment:
This is potentially changing other owner references as well? I would have expected you to only change the one we set here. I don't think it matters much, I was just wondering if you had a reason for it?

Reply:
I'm not sure in which cases we may have several owner references in the template, but my guess is that as long as there is at least one `BlockOwnerDeletion` set to true then the creation will fail.
So we must take care of all the owner references ?